anvill/src/Lifters/FunctionLifter.cpp

```diff
@@ -988,18 +1013,20 @@ void FunctionLifter::VisitInstructions(uint64_t address) {
     //            back to the entrypoint of our function. In this case, treat it
     //            like a tail-call.
     if (inst_addr != func_address || from_addr) {
-      auto maybe_decl = type_provider.TryGetFunctionType(inst_addr);
-      if (maybe_decl) {
-        llvm::Function *const other_decl = DeclareFunction(*maybe_decl);
+
+      auto maybe_decl = TryGetTargetFunctionType(inst_addr);
+      if (maybe_decl.has_value()) {
+        auto decl = maybe_decl.value();
+        llvm::Function *const other_decl = DeclareFunction(decl);
 
         if (const auto mem_ptr_from_call =
-                TryCallNativeFunction(inst_addr, other_decl, block)) {
+                TryCallNativeFunction(decl.address, other_decl, block)) {
           llvm::ReturnInst::Create(llvm_context, mem_ptr_from_call, block);
           continue;
         }
 
         LOG(ERROR) << "Failed to call native function at " << std::hex
-                   << inst_addr
+                   << decl.address
```

Comment:
Change to: `"Failed to call native function " << std::hex << decl.address << " at " << inst_addr << " via ...`